client/scripts/controllers/MonitoringController.js

```diff
@@ -74,7 +78,11 @@ require('../app').controller('MonitoringController', /* @ngInject */function ($s
   }
 
   controller.updateFilter = function () {
-    var filter = _.mapValues(controller.filter, function (value) {
+    var filter = _.mapValues(controller.filter, function (value, key) {
+      if (key === 'threat' || key === 'category') {
```

Comment:
instead of depending on the key, can check for `instanceof Nomenclature`
```suggestion
      if (value instanceof Nomenclature) {
```